packages/itwinui-react/src/core/ThemeProvider/ThemeProvider.tsx

```diff
@@ -96,19 +115,23 @@ export const ThemeProvider = React.forwardRef((props, forwardedRef) => {
     theme: themeProp = 'inherit',
     children,
     themeOptions,
+    portalContainer: portalContainerProp,
     ...rest
   } = props;
 
-  const portalContainerRef = React.useRef<HTMLDivElement>(null);
+  const [portalContainerState, setPortalContainerState] = React.useState(
+    portalContainerProp || null,
```

Comment:
Is this initialization necessary? Is it not okay if it remains null/undefined when a `portalContainerProp` is passed. Because anyways portalContainer is what is really being passed to the context.

Reply:
i think you're right in this case, but in other future cases of `useUncontrolledState`, it could be useful. plus it ensures the correct types are used for the return values of `useState`

https://github.com/iTwin/iTwinUI/blob/05af7c3dd536f090aafeb808f5423c0f1a556ea6/packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx#L81-L82

Reply:
Superseded by https://github.com/iTwin/iTwinUI/pull/1477/commits/512da81a36f108e886528ecb3160f6a0a85d400e#r1291543460